autopush/db.py

```diff
@@ -853,3 +861,132 @@ def clear_node(self, item):
             return True
         except ConditionalCheckFailedException:
             return False
+
+
+@attrs
+class DatabaseManager(object):
+    """Provides database access"""
+
+    storage = attrib()  # type: Storage
+    router = attrib()   # type: Router
+
+    metrics = attrib()  # type: IMetrics
+
+    message_tables = attrib(default=Factory(dict))  # type: Dict[str, Message]
+    current_msg_month = attrib(default=None)        # type: Optional[str]
+    current_month = attrib(default=None)            # type: Optional[int]
+
+    _message_prefix = attrib(default="message")  # type: str
+
+    @classmethod
+    def from_settings(cls, settings):
```

Comment:
we get a light weight `__init__` automatically w/ @attrs (some tests use it). these might change around a bit though in a later commit, re the deferred initialization comments above